src/errors/join.go

```diff
@@ -10,6 +10,8 @@ package errors
 // The error formats as the concatenation of the strings obtained
 // by calling the Error method of each element of errs, with a newline
 // between each string.
+//
+// A non-nil error returned by Join implements the Unwrap() []error method.
```

Comment:
Would it be appropriate to be more explicit here, and say something about "the returned error can not be unwrapped by [Unwrap]" ?